File: backend/app/routes/games.py
from fastapi import APIRouter
from app.controllers import games as games_controller
from app.core.constants import STATIC_FILE

app = APIRouter()



@app.get("/games")
async def get_game():
    """Obtiene información completa del partido"""
    return await games_controller.get_game_data(STATIC_FILE)


@app.get("/games/summary")
async def get_game_summary():
    """Obtiene un resumen del partido"""
    return await games_controller.get_game_summary(STATIC_FILE)


@app.get("/games/teams")
async def get_teams():
    """Obtiene información de los equipos del partido"""
    return await games_controller.get_game_teams(STATIC_FILE)


@app.get("/games/timeline")
async def get_timeline():
    """Obtiene el timeline (fechas y horas clave) del partido"""
    return await games_controller.get_match_timeline(STATIC_FILE)

File: frontend/src/api/games.ts
/**
 * Games API
 * 
 * Funciones para obtener información de partidos
 */

import { apiClient } from './client';
import type { GameData, GameSummary, GameTeams, GameTimeline } from '@/types';

/**
 * Obtiene la información completa del partido
 */
export const getGameData = async (): Promise<GameData> => {
  const response = await apiClient.get<GameData>('/games');
  return response.data;
};

/**
 * Obtiene un resumen del partido
 */
export const getGameSummary = async (): Promise<GameSummary> => {
  const response = await apiClient.get<GameSummary>('/games/summary');
  return response.data;
};

/**
 * Obtiene información de los equipos del partido
 */
export const getGameTeams = async (): Promise<GameTeams> => {
  const response = await apiClient.get<GameTeams>('/games/teams');
  return response.data;
};

/**
 * Obtiene el timeline del partido (fechas y horas clave)
 */
export const getGameTimeline = async (): Promise<GameTimeline> => {
  const response = await apiClient.get<GameTimeline>('/games/timeline');
  return response.data;
};
